File: emulinker/src/main/java/org/emulinker/kaillera/model/event/GameTimeoutEvent.java
package org.emulinker.kaillera.model.event;

import org.emulinker.kaillera.model.*;

public class GameTimeoutEvent implements GameEvent {
  private KailleraGame game;
  private KailleraUser user;
  private int timeoutNumber;

  public GameTimeoutEvent(KailleraGame game, KailleraUser user, int timeoutNumber) {
    this.game = game;
    this.user = user;
    this.timeoutNumber = timeoutNumber;
  }

  public String toString() {
    return "GameTimeoutEvent";
  }

  public KailleraGame getGame() {
    return game;
  }

  public KailleraUser getUser() {
    return user;
  }

  public int getTimeoutNumber() {
    return timeoutNumber;
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/event/GameTimeoutEvent.kt
package org.emulinker.kaillera.model.event

import org.emulinker.kaillera.model.KailleraGame
import org.emulinker.kaillera.model.KailleraUser

data class GameTimeoutEvent(
    override val game: KailleraGame, val user: KailleraUser, val timeoutNumber: Int
) : GameEvent {
  override fun toString(): String {
    return "GameTimeoutEvent"
  }
}
